api/client/src/main/java/org/projectnessie/client/auth/oauth2/OAuth2Client.java

```diff
@@ -171,34 +170,32 @@ public void close() {
     }
   }
 
+  private void wakeUp(Instant now) {
+    LOGGER.debug("Waking up...");
+    Duration delay = nextTokenRefresh(getCurrentTokens(), now, Duration.ZERO);
+    if (delay.compareTo(MIN_REFRESH_DELAY) < 0) {
+      LOGGER.debug("Refreshing tokens immediately");
+      renewTokens();
```

Comment:
The goal is to isolate this (synchronous) call to `renewTokens` in a method that is only executed in a wake-up situation.